resharper/resharper-unity/src/UnityVersion.cs

```diff
@@ -132,12 +134,50 @@ public static Version Parse(string input)
             Version version = null;
             if (match.Success)
             {
-                var type = Convert.ToInt32(groups["type"].Value+groups["revision"], 16);
-                version = Version.Parse($"{groups["major"].Value}.{groups["minor"].Value}.{groups["build"].Value}.{type}");
+                var typeWithRevision = "0";
+                try
+                {
+                    var typeChar = groups["type"].Value.ToCharArray()[0];
+                    var shiftedChar = 16 + typeChar; // Because `f1` = `1021` and `b10` = `9810`, which will break sorting
+                    var revision = Convert.ToInt32(groups["revision"].Value);
+                    typeWithRevision = shiftedChar.ToString("D3") + revision.ToString("D3");
+                }
+                catch (Exception e)
+                {
+                    ourLogger.Error($"Unable to parse part of version. type={groups["type"].Value} revision={groups["revision"].Value}", e);
+                }
+
+                version = Version.Parse($"{groups["major"].Value}.{groups["minor"].Value}.{groups["build"].Value}.{typeWithRevision}");
+            }
+            else
+            {
+                ourLogger.Error($"Unable to version. input={input}");    
             }
 
             return version;
         }
+        
+        public static string VersionToString(Version version)
+        {
+            var type = string.Empty;
+            var rev = string.Empty;
+            try
+            {
+                var revisionString = version.Revision.ToString(); // first 3 is char, next 1+ ones - revision
+                if (revisionString.Length > 3)
+                {
+                    var charValue = Convert.ToInt32(revisionString.Substring(0, 3)) - 16;
```

Comment:
You can't use 3 characters here. `b` is 98. You'll need to do the opposite of `<< 16`:

```
char charValue = (char)(version.Revision >> 16) & 0xFF;
```